src/hooks/useRandomDrink.ts

```diff
@@ -0,0 +1,24 @@
+import { type RandomDrink } from '../utils/APIResponsesTypes';
+import { useEffect, useState } from 'react';
+
+const useRandomDrink = (): RandomDrink | undefined => {
+  const [data, setData] = useState<RandomDrink>();
+
+  useEffect(() => {
+    const fetchRandomDrink = async (): Promise<void> => {
```

Comment:
you use this [...]  const result = await fetch(`${process.env.NEXT_PUBLIC_URL}/lookup.php?i=${id}`);
        const response = await result.json(); [...] multiple times maybe create hook useFetch?